pkg/scrypto/scrypto_spec.gobra

```diff
@@ -28,12 +28,15 @@ import sl "github.com/scionproto/scion/verification/utils/slices"
 // ghost function.
 ghost
 requires  acc(sl.Bytes(key, 0, len(key)), _)
+requires  len(key) > 0
 decreases _
 pure func ValidKeyForHash(key []byte) bool
```

Comment:
We do we not specify any criteria on the length of key? I looked at the implementation of `InitMac()` and it seems that it will only return a new hash if the length of key is either 16, 24, or 32 bytes. Hence requiring `len(key) > 0` cannot cause any problems?

Reply:
Well, while that it is true, this restriction is not documented anywhere; as such, this might be an implementation detail, which should not be leaked in the specification. Passing a key of size 0 sounds non-sensical, but that is only one of the criteria that one might think for this.

Why is adding this restriction here useful? If it is not, then I would drop the restriction, considering that this case is still correctly handled by the implementation, which returns a non-nil error in that case

Reply:
We need to ensure that either `len(verScionTmp.UBuf) > 0` or `len(d.key) > 0` in `newPacketProcessor() `(dataplane.go) holds. We know from `NewSerializeBuffer()` that `len(verScionTmp.UBuf) == 0`. This means the condition `len(d.key) > 0` must hold, passed via `dataplane.Mem()` in `run()`. The minimal required change is to add `len(d.key) > 0` to `dataplane.Mem()`. So adding it to `ValidKeyForHash()` is optional. I just thought it would make it clearer why we need it in `dataplane.Mem()`. If you think it is better to remove it let me know.

Reply:
No, unfortunately this is still not strong enough. So far I don't see a way without requiring `len(d.key) > 0` in `dataplane.Mem()`.